npm/src/upload/index.ts

```diff
@@ -43,6 +50,17 @@ export class FileUploader {
   ) => {
     window.addEventListener("beforeunload", pageUnloadAction)
 
+    const intervalId = setInterval(() => {
+      if (idleSessionTimeoutAboutToExpire(this.keycloak)) {
+        //Allow for refreshing the token before the refresh token expires
+        //Keycloak documentation: https://www.keycloak.org/docs/latest/server_admin/#user-session-management
+        /*
+         SSO Session Idle: ... The idle timeout is reset by a client requesting authentication or by a refresh token request. ...
+        */
+        refreshOrReturnToken(this.keycloak)
```

Comment:
Just to check - does this exchange the refresh token for a new set of tokens? As opposed to launching an iframe, I mean.

Reply:
As far as I can tell debugging through, and looking at the response from the request, it provides a new set of tokens with different expiry times. The iframe doesn't appear to be touched.